src/scenes/Projects/Members/Create/CreateProjectMember.tsx

```diff
@@ -0,0 +1,138 @@
+import { Container, makeStyles } from '@material-ui/core';
+import React from 'react';
+import { FormSpy } from 'react-final-form';
+import { Except } from 'type-fest';
+import { displayRole, Role } from '../../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../../components/Dialog/DialogForm';
+import { FieldGroup, SubmitError } from '../../../../components/form';
+import { AutocompleteField } from '../../../../components/form/AutocompleteField';
+import { UserField, UserLookupItem } from '../../../../components/form/Lookup';
+import { Nullable } from '../../../../util';
+import { ProjectMembersDocument } from '../List/ProjectMembers.generated';
+import { useCreateProjectMemberMutation } from './CreateProjectMember.generated';
+
+interface DialogValues {
+  projectMember: {
+    userId: Nullable<UserLookupItem>;
+    projectId: string;
+    roles?: Role[];
+  };
+}
+
+type CreateProjectMemberProps = Except<
+  DialogFormProps<DialogValues>,
+  'onSubmit' | 'initialValues'
+> & {
+  projectId: string;
+};
+
+const Roles: Role[] = [
```

Comment:
We don't have a standard of where to throw these constants.  We can either keep it in the component for 1 time uses or have a constants directory in our api/utils/etc

Reply:
If we had a `RoleField` it would make since there ;)

I'm thinking about auto generating these lists from the api schema, so we can use them at run time. But that's for another time. It's fine here or in api/ somewhere

Reply:
Sounds good, i'll look into the `RoleField` as I do the edit